src/components/SearchCard/SearchCard.tsx

```diff
@@ -0,0 +1,47 @@
+import { Form, FormLayout, TextField, Button, Card } from "@shopify/polaris";
+import React, { useState } from "react";
+import { useHistory } from "react-router-dom";
+import { useTranslation } from "react-i18next";
+import { postalCodeIsValid, postalCodeToBrowserFormat } from "../../utils";
+
+export function SearchCard() {
+  const { t } = useTranslation();
+  const [shouldShowInvalidPostal, setShouldShowInvalidPostal] = useState(false);
+  const [postalCode, setPostalCode] = useState("");
+  const history = useHistory();
+
+  const handleSubmit = () => {
+    if (postalCodeIsValid(postalCode)) {
+      history.push(`/search/${postalCodeToBrowserFormat(postalCode)}`);
+    } else {
+      setShouldShowInvalidPostal(true);
+    }
+  };
+
+  const invalidPostalCodeMessage = shouldShowInvalidPostal
+    ? t("invalidpostal")
+    : undefined;
+
+  return (
+    <>
+      <Card>
+        <Card.Section>
+          <Form onSubmit={handleSubmit}>
```

Comment:
Hmm, there is a `role="search"` attribute value we could apply to search forms as an assistive cue, but it looks like the `Form` component doesn't support aria props... The `Form` props should probably extend from `IntrinsicAttributes` so as to support attributes assignable to HTML elements.